Cloud-controller-manager won't build for out-of-tree external providers

**What happened**:

panic: failed to find flag in global flagset (flag): cloud-provider-gce-lb-src-cidrs

goroutine 1 [running]:
k8s.io/component-base/cli/globalflag.Register(0xc0000ba900, 0x17df195, 0x1f)
	/home/neil/Go/pkg/mod/k8s.io/component-base@v0.0.0-20190405092941-79a63e28c077/cli/globalflag/globalflags.go:66 +0x1a9
k8s.io/kubernetes/cmd/cloud-controller-manager/app.NewCloudControllerManagerCommand(0xc00008e150)
	/home/neil/Go/pkg/mod/k8s.io/kubernetes@v1.14.0/cmd/cloud-controller-manager/app/controllermanager.go:91 +0x438
main.main()
	/home/neil/Dev/brightbox-cloud-controller-manager/main.go:39 +0x90

**What you expected to happen**:

The controller should run and stop with a '--cloud-provider cannot be empty' message

**How to reproduce it (as minimally and precisely as possible)**:

Follow the out of tree instructions here https://kubernetes.io/docs/tasks/administer-cluster/developing-cloud-controller-manager/

Comment out the _ "k8s.io/kubernetes/pkg/cloudprovider/providers" import line and build the cloud-controller-manager

**Anything else we need to know?**:

This worked fine prior to 1.14.0 

**Environment**:
- Kubernetes version (use `kubectl version`):
lient Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:53:57Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:
Brightbox
- OS (e.g: `cat /etc/os-release`):
Ubuntu 18.04.2 LTS
- Kernel (e.g. `uname -a`):
Linux bionic-vm 4.15.0-47-generic #50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linu
- Install tools:
kubeadm
- Others:

--- a/cmd/cloud-controller-manager/app/controllermanager.go
+++ b/cmd/cloud-controller-manager/app/controllermanager.go
@@ -18,6 +18,7 @@ package app
 
 import (
 	"context"
+	"flag"
 	"fmt"
 	"net/http"
 	"os"
@@ -86,9 +87,12 @@ the cloud specific control loops shipped with Kubernetes.`,
 	namedFlagSets := s.Flags(KnownControllers(), ControllersDisabledByDefault.List())
 	verflag.AddFlags(namedFlagSets.FlagSet("global"))
 	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
-	// hoist this flag from the global flagset to preserve the commandline until
-	// the gce cloudprovider is removed.
-	globalflag.Register(namedFlagSets.FlagSet("generic"), "cloud-provider-gce-lb-src-cidrs")
+
+	if flag.CommandLine.Lookup("cloud-provider-gce-lb-src-cidrs") != nil {
+		// hoist this flag from the global flagset to preserve the commandline until
+		// the gce cloudprovider is removed.
+		globalflag.Register(namedFlagSets.FlagSet("generic"), "cloud-provider-gce-lb-src-cidrs")
+	}
 	for _, f := range namedFlagSets.FlagSets {
 		fs.AddFlagSet(f)
 	}

